invoke-ai/InvokeAI
# feat(ui): add new session functionality

## Summary

[feat(ui): add new gallery/canvas session buttons to queue actions menu](https://github.com/invoke-ai/InvokeAI/commit/20d731aab3e4539b37d3126325c05a6c5fd24d41)

A new "session" just means to reset most settings to default values, excluding model.

There are a few things that need to be reset:
- Parameters state, except for models and things dependent on model selection (like VAE precision)
- Canvas state, except for the `modelBase`, which is dependent on the model selection
- Canvas staging area state
- LoRAs state
- HRF state
- Style presets state

We also select the canvas tab.

For new gallery sessions, we:
- Open the image viewer
- Set the right panel tab to `gallery`

And for new canvas sessions, we:
- Close the image viewer
- Set the right panel tab to `layers`

## Related Issues / Discussions

offline discussion

## QA Instructions

Use these buttons to create a new session:
![image](https://github.com/user-attachments/assets/fb633e50-5818-4a8f-bcbb-8f8c236eefbd)

You'll be prompted to confirm:
![image](https://github.com/user-attachments/assets/4a1a2d6c-0243-43b5-b3a2-61bbd54f0051)

If you select `don't ask me again`, you can re-enable the confirmations in the settings modal.

All generation settings/state, besides models and model-dependent state, should be reset, as described in the summary above.

## Merge Plan

n/a

## Checklist
